config_proc_macro/src/style_edition.rs

```diff
@@ -0,0 +1,355 @@
+        path.segments.push(varient.ident.clone().into());
+        syn::ExprPath {
+            attrs: vec![],
+            qself: None,
+            path,
+        }
+    }
+
+    /// Set the style edition based on the the annotated attribute
+    /// For example:
+    /// ```ignore
+    /// #[style_edition]
+    /// enum Example {
+    ///     #[se_default] // <-- Default style edition
+    ///     A,
+    ///     #[se_2018] // <-- Explicit override for StypeEdition2018
+    ///     B,
+    /// }
+    /// ```
+    fn set_defaults_by_enum_variant_attr(&mut self, en: &syn::ItemEnum) {
+        for varient in en.variants.iter() {
+            for attr in varient.attrs.iter() {
+                if se_default(attr) {
+                    self.default.replace(Self::enum_expr_path(varient, en).into());
+                    break;
+                } else if se_2015(attr) {
+                    self.se2015.replace(Self::enum_expr_path(varient, en).into());
+                    break;
+                } else if se_2018(attr) {
+                    self.se2018.replace(Self::enum_expr_path(varient, en).into());
+                    break;
+                } else if se_2021(attr) {
+                    self.se2021.replace(Self::enum_expr_path(varient, en).into());
+                    break;
+                } else if se_2024(attr) {
+                    self.se2024.replace(Self::enum_expr_path(varient, en).into());
+                    break;
+                }
+            }
+        }
+    }
+
+    /// Set the style edition based on the rhs of the assignment in the attribute
+    /// e.g. `#[style_edition(true, se_2015=false)]`
+    fn set_by_assignment(&mut self, assignment: &syn::ExprAssign) -> syn::Result<()> {
+        match assignment.left.as_ref() {
+            syn::Expr::Path(expr) => {
+                let se2015 = syn::Ident::new("se_2015", Span::call_site());
+                let se2018 = syn::Ident::new("se_2018", Span::call_site());
+                let se2021 = syn::Ident::new("se_2021", Span::call_site());
+                let se2024 = syn::Ident::new("se_2024", Span::call_site());
+                let ident = expr
+                    .path
+                    .segments
+                    .first()
+                    .map(|segment| segment.ident.clone())
+                    .expect("should be at least one ident");
+
+                if ident == se2015 {
+                    self.se2015.replace(*assignment.right.clone());
+                    return Ok(());
+                } else if ident == se2018 {
+                    self.se2018.replace(*assignment.right.clone());
+                    return Ok(());
+                } else if ident == se2021 {
+                    self.se2021.replace(*assignment.right.clone());
+                    return Ok(());
+                } else if ident == se2024 {
+                    self.se2024.replace(*assignment.right.clone());
+                    return Ok(());
+                }
```

Comment:
seems repetitive. Would this be easier to read?

```suggestion
                for (edition, field) in [
                    ("se2015", &mut self.se2015),
                    ("se2018", &mut self.se2018),
                    ("se2021", &mut self.sf2021),
                    ("se2024", &mut self.se2024),
                ]
                {
                    if ident == edition {
                        field.replace(*assignment.right.clone());
                        return Ok(());
                    }
                }
```